nc/source/physics/IntersectionQueries.cpp

```diff
@@ -0,0 +1,300 @@
+        auto maxDot_v = XMVector3Dot(maxVertex_v, direction_v);
+
+        for(size_t i = 1u; i < vertices.size(); ++i)
+        {
+            auto vertex_v = XMLoadVector3(&vertices[i]);
+            auto dot_v = XMVector3Dot(vertex_v, direction_v);
+            if(XMVector3Greater(dot_v, maxDot_v))
+            {
+                maxDot_v = dot_v;
+                maxVertex_v = vertex_v;
+            }
+        }
+
+        return maxVertex_v;
+    }
+
+    XMVECTOR MinkowskiSupport(const BoundingVolume& collider, FXMVECTOR direction_v)
+    {
+        return std::visit([&direction_v](auto&& bv)
+        {
+            return MinkowskiSupport(bv, direction_v);
+        }, collider);
+    }
+
+    bool RefinePoint(Simplex& simplex, Vector3& direction)
+    {
+        direction = -simplex[0];
+        return false;
+    }
+
+    bool RefineLine(Simplex& simplex, Vector3& direction)
+    {
+        auto ao = -1.0f * simplex[0];
+        auto ab = simplex[1] + ao;
+
+        if(SameDirection(ab, ao))
+        {
+            direction = TripleCrossProduct(ab, ao, ab);
+        }
+        else
+        {
+            simplex.ToPoint<0u>();
+            direction = ao;
+        }
+
+        return false;
+    }
+
+    bool RefineTriangle(Simplex& simplex, Vector3& direction)
+    {
+        auto ao = -1.0f * simplex[0];
+        auto ab = simplex[1] + ao;
+        auto ac = simplex[2] + ao;
+        auto abc = CrossProduct(ab, ac);
+
+        if(SameDirection(CrossProduct(abc, ac), ao))
+        {
+            if(SameDirection(ac, ao))
+            {
+                simplex.ToLine<0u, 2u>();
+                direction = TripleCrossProduct(ac, ao, ac);
+            }
+            else
+            {
+                simplex.ToLine<0u, 1u>();
+                return RefineLine(simplex, direction);
+            }
+        }
+        else
+        {
+            if(SameDirection(CrossProduct(ab, abc), ao))
+            {
+                simplex.ToLine<0u, 1u>();
+                return RefineLine(simplex, direction);
+            }
+            else
+            {
+                if(SameDirection(abc, ao))
+                {
+                    direction = abc;
+                }
+                else
+                {
+                    simplex.ToTriangle<0u, 2u, 1u>();
+                    direction = -1.0f * abc;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    bool RefineTetrahedron(Simplex& simplex, Vector3& direction)
+    {
+        auto ao = -1.0f * simplex[0];
+        auto ab = simplex[1] + ao;
+        auto ac = simplex[2] + ao;
+        auto ad = simplex[3] + ao;
+        auto abc = CrossProduct(ab, ac);
+        auto acd = CrossProduct(ac, ad);
+        auto adb = CrossProduct(ad, ab);
+
+        if(SameDirection(abc, ao))
+        {
+            simplex.ToTriangle<0u, 1u, 2u>();
+            return RefineTriangle(simplex, direction);
+        }
+        else if(SameDirection(acd, ao))
```

Comment:
Don't need the else if's here vecause you are returning. 

Can just be:
if(x)
{ return; }

if (y)
{ return; }

etc

Reply:
It will make the code a little neater